src/v/transform/logging/event.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "base/seastarx.h"
+#include "model/transform.h"
+#include "utils/named_type.h"
+
+#include <seastar/core/sstring.hh>
+
+#pragma once
+
+namespace transform::logging {
+using transform_log_level
+  = named_type<ss::log_level, struct transform_log_level_tag>;
```

Comment:
Is this extra layer useful?

We could define our own enum and just translate or just use `ss::log_level` directly.

Reply:
>is this extra layer useful

Only for rjson. I wanted to avoid defining overloads for generally applicable types. Definitely doesn't need to be in the interface though, and might be less relevant now that `event` has its own compilation unit.

Reply:
Removed this